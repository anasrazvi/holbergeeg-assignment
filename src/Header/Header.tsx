import {observer} from 'mobx-react';
import {storeProps} from '../Types/Interfaces';
import HeaderCSS from './Header.module.css';

export const Header: React.FC<storeProps> = observer(({store}) => {
    return (
      <div className={HeaderCSS.header}>
        {store && store.metaData && store.metaData.columnsMetaInformation && store.metaData.columnsMetaInformation.map((col: any, idx: number) => {  
          if(col.isVisible){
            return(
              <div className="col" key={idx}><strong>{col.name?.primaryValue || col.name?.secondaryValue}</strong></div>
            )
          }}
        )}
      </div>
    ) 
  })