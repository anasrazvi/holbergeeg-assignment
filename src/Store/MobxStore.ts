import { autorun, makeAutoObservable } from "mobx";
import {columnsMeta, ColumnsMetaInformation, StudyData, DataEntity, StudyDataMapped } from '../Types/Interfaces';

export class ObservableStoreImpl {
    //Initialize the store
    metaData: ColumnsMetaInformation = {
        columnsMetaInformation: []
    }
    studyData: StudyData = {
      data: []
    }
    studyDataMapped: StudyDataMapped = {
      data: new Map<number, DataEntity>()
    }
    isLoaded: boolean = true;
    constructor() {
      makeAutoObservable(this)
      autorun(() => console.log(this.metaData));
    }
  
    //Setters
    setMetaDataSuccess(data: columnsMeta[]) {
      data.sort((a,b) => {return (a.order || 0) - (b.order || 0)})
      this.metaData.columnsMetaInformation = data;
    }

    setStudyDataSuccess(data: DataEntity[]){
      /*const mappedData: Map<number, DataEntity> = new Map(data.map(study => [study.studyId?.primaryValue || study.studyId?.secondaryValue, study] as [number, DataEntity]))
      mappedData.forEach((value: DataEntity, key: number) => {
        this.studyDataMapped.data.set(key, value);
      })*/
      this.studyData.data = data;
    }
    
    setVisibility(colIndex: number, isVisible: boolean){
      if(this.metaData.columnsMetaInformation){
        this.metaData.columnsMetaInformation[colIndex].isVisible = isVisible; 
      }
    }

    setLoader(isLoading: boolean){
      this.isLoaded = isLoading;
    }

    //Getters
    get Loader(){
      return this.isLoaded;
    }

    get Metadata() {
       return { metaData: this.metaData.columnsMetaInformation }
    }

    get StudyData(){
      return { 
        studyData: this.studyData.data
      }
    }
  }
  
  export const observableStore = new ObservableStoreImpl();