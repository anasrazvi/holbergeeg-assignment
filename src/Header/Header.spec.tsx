import renderer from 'react-test-renderer';
import {Header} from '../Header/Header';
import { ObservableStoreImpl } from '../Store/MobxStore';
import {data} from '../StudyData';

const store = new ObservableStoreImpl();
store?.setMetaDataSuccess(data.GetMetadata().columnsMetaInformation);
store?.setStudyDataSuccess(data.GetStudyList().data);

describe("Grid Header Component", () => {
    test('renders grid header', () => {
      const header = renderer.create(<Header store={store}></Header>).toJSON();
      expect(header).toMatchSnapshot();
    });
});