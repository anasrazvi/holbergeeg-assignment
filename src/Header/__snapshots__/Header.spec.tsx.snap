// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid Header Component renders grid header 1`] = `
<div
  className="header"
>
  <div
    className="col"
  >
    <strong>
      Study start
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Study length
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Study type
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Study status
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Name
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Patient ID
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Date of birth
    </strong>
  </div>
  <div
    className="col"
  >
    <strong>
      Age at the time of study
    </strong>
  </div>
</div>
`;
